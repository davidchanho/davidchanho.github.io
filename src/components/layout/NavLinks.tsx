import NavLink from "components/layout/NavLink";
import React from 'react';

const links = [
    { link: 'about', name: 'About' },
    { link: 'education', name: 'Education' },
    { link: 'work', name: 'Work' },
    { link: 'projects', name: 'Projects' },
    { link: 'tech', name: 'Favorite Tech' },
    { link: "contacts", name: 'Contact Me' }
]

function NavLinks() {
    return (
        <div>
            {links.map(({ link, name }, index) => (
                <NavLink key={`nav-link-${index}`} link={link} name={name} />
            ))}
        </div>
    )
}

export default NavLinks
