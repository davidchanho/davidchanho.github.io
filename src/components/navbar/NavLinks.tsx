import React from 'react';
import { Link } from 'react-scroll';
import styled from 'styled-components';

const NavLink = styled(Link)`
 cursor: pointer;
 margin-right: 15px;
`;

function NavLinks() {
    return (
        <div className="ml-auto" >
            <NavLink activeClass="active" to="about" spy={true} hashSpy={true} smooth={true} duration={500}>
                About
            </NavLink>
            <NavLink activeClass="active" to="projects" spy={true} hashSpy={true} smooth={true} duration={500}>
                Projects
            </NavLink>
            <NavLink activeClass="active" to="contactForm" spy={true} hashSpy={true} smooth={true} offset={-200} duration={500}>
                Contact Me
            </NavLink>
        </div>
    )
}

export default NavLinks
