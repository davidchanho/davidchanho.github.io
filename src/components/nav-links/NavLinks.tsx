import { NavLink } from 'components/nav-links/NavLinks.styles';
import React from 'react';

function NavLinks() {
    return (
        <div className="ml-auto" >
            <NavLink activeClass="active" to="about" spy={true} hashSpy={true} smooth={true} duration={500}>
                About
            </NavLink>
            <NavLink activeClass="active" to="projects" spy={true} hashSpy={true} smooth={true} duration={500}>
                Projects
            </NavLink>
            <NavLink activeClass="active" to="blog" spy={true} hashSpy={true} smooth={true} duration={500}>
                Blog
            </NavLink>
            <NavLink activeClass="active" to="contactForm" spy={true} hashSpy={true} smooth={true} offset={-200} duration={500}>
                Contact Me
            </NavLink>
        </div>
    )
}

export default NavLinks
