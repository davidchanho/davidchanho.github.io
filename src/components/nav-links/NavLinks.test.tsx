import React from "react";
import renderer from "react-test-renderer";
import NavLinks from ".";

describe("NavLinks", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NavLinks />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
