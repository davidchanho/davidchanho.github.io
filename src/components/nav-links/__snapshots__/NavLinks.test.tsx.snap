// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders correctly 1`] = `
<div
  className="ml-auto"
>
  <a
    className="sc-bdfBwQ cOVWtI"
    onClick={[Function]}
  >
    About
  </a>
  <a
    className="sc-bdfBwQ cOVWtI"
    onClick={[Function]}
  >
    Projects
  </a>
  <a
    className="sc-bdfBwQ cOVWtI"
    onClick={[Function]}
  >
    Blog
  </a>
  <a
    className="sc-bdfBwQ cOVWtI"
    onClick={[Function]}
  >
    Contact Me
  </a>
</div>
`;
